### PR TITLE
Support multi-doc for subscriptions

### PR DESCRIPTION
**Pull request by dvoytenko**
Depends on #14356.

### CODE REVIEW

File: extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js [hunk 1]
```diff
@@ -70,7 +71,7 @@ export class GoogleSubscriptionsPlatform {
     this.serviceAdapter_ = serviceAdapter;
     /** @private @const {!ConfiguredRuntime} */
     this.runtime_ = new ConfiguredRuntime(
-        ampdoc.win,
+        new DocImpl(ampdoc),
```
Comment (prateekbh) [inline extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js]: I guess a small test that we're not using ampdoc directly anymore should be needed here

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js [hunk 2]
```diff
@@ -104,7 +105,8 @@ export class SubscriptionService {
    */
   initialize_() {
     if (!this.initialized_) {
-      const pageConfigResolver = new PageConfigResolver(this.ampdoc_.win);
+      const doc = new DocImpl(this.ampdoc_);
+      const pageConfigResolver = new PageConfigResolver(doc);
```
Comment (prateekbh) [inline extensions/amp-subscriptions/0.1/amp-subscriptions.js]: same

File: extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js (hunk 1 above)
Comment (dvoytenko) [reply extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js]: Done. I wish we could stub constructors :(

File: extensions/amp-subscriptions/0.1/amp-subscriptions.js (hunk 2 above)
Comment (dvoytenko) [reply extensions/amp-subscriptions/0.1/amp-subscriptions.js]: Done

**Pull request merged**